query/README.md

```diff
@@ -0,0 +1,146 @@
+The most simple way that users can use the KeenQueryClient to send queries is as follows. These methods take only the required query parameters as input, and the user receives a very specific Integer or Double response type. Please note that we strongly encourage users to pass in the Timeframe parameter, but it can be null.
+```java
+Long count = queryClient.count("<event_collection>", new RelativeTimeframe("this_year"));
+Long countUnique = queryClient.countUnique("<event_collection>", "<target_property>", new AbsoluteTimeframe("2012-08-13T19:00:00.000Z","2015-06-07T19:00:00.000Z"));
+Double minimum = queryClient.minimum("<event_collection>", "<target_property>", new RelativeTimeframe("this_year"));
+Double maximum = queryClient.maximum("<event_collection>", "<target_property>", new RelativeTimeframe("this_year"));
+Double average = queryClient.average("<event_collection>", "<target_property>", new RelativeTimeframe("this_year"));
+Double median = queryClient.median("<event_collection>", "<target_property>", new RelativeTimeframe("this_year"));
+Double percentile = queryClient.percentile("<event_collection>", "<target_property>", new RelativeTimeframe("this_year"));
+Double sum = queryClient.sum("<event_collection>", "<target_property>", new RelativeTimeframe("this_year"));
+```
+The exceptions are Select Unique, Extraction, Funnel, and Multi-Analysis queries which are more a little more complicated and are mentioned below (TODO: Mention below!!!).
+
+### Advanced
+Alternatively, users can use optional parameters to send queries. However, because the result may be different depending on the optional parameters, the return value is a QueryResult. The user is expected to verify the expected return type for the query, given the parameters entered.
+```java
+Query query = new QueryBuilder(QueryType.COUNT_RESOURCE)
+        .withEventCollection("<event_collection>")
+        .build();
+QueryResult result = queryClient.execute(query, new RelativeTimeframe("this_month"));
+if (result.isLong()) {
+	long countValue = result.longValue();
+	// do something with countValue
+}
+```
+
+Some special cases are when filters "Group By" and "Inteval" are specified, as well as the Select Unique query.
+
+For Select Unique queries, the user gets a list of unique values, given the target property. Therefore, the QueryResult will be a list of unique property values. The QueryResult type only supports Integer, Double, String, and List values; therefore, if the property value is not one of the aforementioned types, then you may not be able to access that value.
+
+``` java
+Query query = new QueryBuilder(QueryType.SELECT_UNIQUE_RESOURCE)
+        .withEventCollection("<event_collection>")
+        .withTargetProperty("click-number")
+        .build();
+QueryResult result = queryClient.execute(query, new RelativeTimeframe("this_month"));
+if (result.isListResult()) {
+	ArrayList<QueryResult> listResults = result.getListResults();
```

Comment:
Nit: There's no reason to make this an `ArrayList` explicitly; just `List<QueryResult>` is a more idiomatic declaration.